pkg/jsonstructure/copy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonstructure
+
+import (
+	"fmt"
+
+	"emperror.dev/errors"
+)
+
```

Comment:
can't the same results be achived by using mapstructure?

Reply:
Sadly, no,
```go
src := map[string]interface{}{ ... }
var dst map[string]interface{}
mapstructure.Decode(src, &dst)
```
will only create a shallow copy of `src` into `dst`.